src/libraries/Microsoft.PowerFx.Interpreter/Functions/StandardErrorHandling.cs

```diff
@@ -301,15 +344,19 @@ public ExpandToSizeResult(string name, IEnumerable<DValue<RecordValue>> rows)
             }
         }
 
-        private static ExpandToSizeResult ExpandToSize(FormulaValue arg, int size)
+        private static ExpandToSizeResult ShrinkToSize(FormulaValue arg, int size)
         {
             if (arg is TableValue tv)
             {
                 var tvType = (TableType)tv.Type;
                 var name = tvType.SingleColumnFieldName;
 
                 var count = tv.Rows.Count();
-                if (count < size)
+                if (count > size)
+                {
+                    return new ExpandToSizeResult(name, tv.Rows.Take(size));
+                }
+                else if (count == 0)
```

Comment:
>else if (count == 0) [](http://example.com/codeflow?start=16&length=20)

In which scenario would we go into this branch? If I understand it correctly, if count == 0 that means we have an empty table - and in this case we don't want to expand it.